src/CommandInvocationParser.cpp

```diff
@@ -18,6 +18,75 @@ namespace cmake::language
     {
       return { std::find_if_not(r.begin, r.end, [](char c) {return std::isspace(c); }), r.end };
     }
+
+    //-----------------------------------------------------------------------------
+    tl::expected<Token, Error> ParserIdentifier(Range r)
+    {
+      auto newRange = IgnoreSpaces(r);
+      auto identifier = Parser<ElementType::Identifier>{}.Parse(newRange);
+      if (identifier)
+      {
+        // Make sure we take into account the spaces
+        identifier->range.begin = r.begin;
+      }
+      return identifier;
+    }
+
+    //-----------------------------------------------------------------------------
+    tl::expected<Token, Error> ParserArguments(Range r)
+    {
+      auto newRange = IgnoreSpaces(r);
+      if (*newRange.begin != '(')
+      {
+        Error err;
+        err.message = "A command invocation should wrap its arguments between parenthesis.";
+        err.context = newRange.begin;
+        return tl::make_unexpected(err);
+      }
+
+      auto args = Parser<ElementType::Arguments>{}.Parse(Range{ newRange.begin + 1, r.end });
+      if (args)
+      {
+        if (args->range.end >= r.end)
+        {
+          Error err;
+          err.message = "Missing parenthesis";
+          err.context = args->range.end;
+          return tl::make_unexpected(err);
+        }
+
+        if (*args->range.end != ')')
+        {
+          Error err;
+          err.message = fmt::format("Expected \')\', got {}", *args->range.end);
+          err.context = args->range.end;
+          return tl::make_unexpected(err);
+        }
+
+        args->range = Range{ r.begin, args->range.end + 1 };
+      }
+      else
+      {
+        if (newRange.begin + 1 == newRange.end)
+        {
+          Error err;
+          err.message = "Missing parenthesis at the end of the command invocation";
+          err.context = newRange.begin + 1;
+          return tl::make_unexpected(err);
+        }
+        if (*(newRange.begin + 1) != ')')
+        {
+          Error err;
+          err.message = fmt::format("Expected \')\', got {}", *(newRange.begin + 1));
+          err.context = newRange.begin + 1;
+          return tl::make_unexpected(err);
+        }
```

Comment:
For the last two error cases, do you see a way to factorize them with the precedent ones which are very similar, while keeping the readability to an ok level ?

Reply:
Yes, I think we should be able to construct a range in the `if/else` and then check for an error / return a result. I will look at it.